admin/config

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash -eu
+
+#
+# BMC config backup/restore.
+#
+
+# prevent modification of executable path
+PATH="/bin:/sbin:/usr/bin:/usr/sbin"
+
+# elevate self privileges
+[[ ${EUID} -eq 0 ]] || exec sudo $(realpath ${BASH_SOURCE[0]}) $@
+
+# list of files to save/restore
+WHITELIST="/var/run/initramfs/whitelist"
+
+# parse command line options
+ACTION="help"
+[[ $# -eq 0 ]] || ACTION="$1"
+case "${ACTION}" in
+  help | -h | --help )
+    echo "Backup or restore OpenBMC configuration."
+    echo "Use: ${0##*/} [OPTION...]"
+    echo "  backup FILE   Save configuration to file"
+    echo "  restore FILE  Restore configuration from file"
+    echo "  reset         Reset configuration to manufacturing default"
+    echo "  -h, --help    Print this help and exit"
+    ;;
+
+  backup )
+    [[ $# -eq 2 ]] || { echo "Invalid arguments" >&2; exit 1; }
+    tar czf "$2" -T ${WHITELIST} &>/dev/null || /bin/true
+    if [[ -f $2 ]]; then
+      echo "Backup created: $2"
+    else
+      echo "Error creating backup: $2" >&2
+      exit 1;
+    fi
+    ;;
+
+  restore )
+    [[ $# -eq 2 ]] || { echo "Invalid arguments" >&2; exit 1; }
+    TMP_DIR="$(mktemp -d)"
+    tar xf "$2" -C ${TMP_DIR}
+    for FILE in $(cat ${WHITELIST}); do
+      [[ -e ${TMP_DIR}${FILE} ]] || continue
```

Comment:
I like this small format of if-then-fi, it is simple, easy to read and fits in one line.
Inverse logic is used because all these scripts are executed with -e option.

I can replace them with long format if you insist.

Reply:
No, I'm fine with oneliners, I'm curious why the logic is inverted. I.e. why `if (expected) {} else { reaction }` instead of `if (!expected) {reaction}`, that is:
```
[[ ! -e ${TMP_DIR}${FILE} ]] && continue
```
looks much more natural to me.

Reply:
Because of `set -e` - bash terminates execution if the first condition fails.

Reply:
Sorry, I forgot about one detail. These conditions have different behavior in script's body and inside functions, so I always use inverse logic to not to be confused.
Fixed.